### PR TITLE
メンター募集ドメインオブジェクトを実装

### PR DESCRIPTION
**Pull request by TakayukiHirano117**
## 概要
メンター募集のエンティティ・値オブジェクトとそのテストを実装しました。

## テストしたこと
- 各テストが通ること
- サンプルユースケースに記載のルールがエンティティ・値オブジェクトで表現されていること

## レビュアーに伝えたいこと
- カテゴリーはプラン集約からも参照されるため、別ディレクトリに分けています。

---

> [!NOTE]
> Implements the `MentorRecruitment` aggregate with supporting value objects and validations, introduces a `categorydm` domain module, and adds a Makefile format target with comprehensive tests.
> 
> - **Domain: mentor recruitment (`src/core/domain/mentor_recruitmentdm/`)**:
>   - **Entity**: `MentorRecruitment` with constructors (`NewMentorRecruitment`, `NewMentorRecruitmentByVal`).
>   - **Value Objects**: `MentorRecruitmentID`, `Title`, `Description`, `ConsultationType`, `ConsultationMethod`, `Budget`, `ApplicationPeriod`, `Status` with validation, equality, and string/time accessors.
> - **Domain: category (`src/core/domain/categorydm/`)**:
>   - **Entity/VOs**: `Category` (with getters, timestamps), `CategoryID`, `CategoryName` with validation/equality.
> - **Tests**:
>   - Add unit tests covering all new entities/VOs and their edge cases.
> - **Tooling**:
>   - Makefile: add `format-docker` target for `go fmt`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 66648d932e6c517884f58bb1d3cc37c64a3a1db4. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: src/core/domain/mentor_recruitmentdm/budget.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
+package mentor_recruitmentdm
+
+import "github.com/cockroachdb/errors"
+
+type Budget struct {
```
Comment (revenue-hack) [inline src/core/domain/mentor_recruitmentdm/budget.go]: これってあえて作り意味って何かありましたか？

recruitmentドメインにプリミティブとして持つでも良いかなと思ったので。

File: src/core/domain/mentor_recruitmentdm/budget_test.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+			assert.Equal(t, tt.min, budget.Min())
+			assert.Equal(t, tt.max, budget.Max())
+		})
+	}
+}
+
+func TestBudget_Equal(t *testing.T) {
+	budget1, _ := mentor_recruitmentdm.NewBudget(3000, 10000)
+	budget2, _ := mentor_recruitmentdm.NewBudget(3000, 10000)
+	budget3, _ := mentor_recruitmentdm.NewBudget(5000, 10000)
```
Comment (revenue-hack) [inline src/core/domain/mentor_recruitmentdm/budget_test.go]: エラーチェック

File: src/core/domain/mentor_recruitmentdm/consultation_method.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
+package mentor_recruitmentdm
+
+import "github.com/cockroachdb/errors"
+
+type ConsultationMethod string
+
+const (
+	chat      ConsultationMethod = "チャット"
+	videoCall ConsultationMethod = "ビデオ通話"
```
Comment (revenue-hack) [inline src/core/domain/mentor_recruitmentdm/consultation_method.go]: これってprivateで持っている意味って何かありますか？
どういうふうに使うのかなと

File: src/core/domain/mentor_recruitmentdm/budget.go (hunk 1 above)
Comment (TakayukiHirano117) [reply src/core/domain/mentor_recruitmentdm/budget.go]: こちらはドメインルールがあるからというのが作った理由になります。
1000円以上の数値であるというルールがサンプルユースケースのnotionに記載されていたのと、
実際にMENTAを触ってみたところ上限が1000000円までだったのでこれは値オブジェクトとした方があとあとルールが変わった時にも改修しやすいし値オブジェクトでドメインルールを表現できていてコードのドキュメント化ができるかと思って作っています。

Comment (revenue-hack) [reply src/core/domain/mentor_recruitmentdm/budget.go]: 全体的な話ですが、全部のフィールドをvo化していますが、全部やる必要はないです。

確かに責務を分離できますが、さほどロジックのないものをvo化すると、ファイル数が増えて逆に管理が煩雑になったりもするので、今回は特に運用をする実装ではないので良いですが、実務ではその辺りは適宜変えていくと良いです。

例えばtitleとかただの文字数制限程度しかないし、使いか回すということも考えづらいので、プリミティブでよかったりします。
それでいうとこれもあまり大きなロジックがないのと、使いまわすこともないのでプリミティブで個人的には良いかなと思います。

Comment (TakayukiHirano117) [reply src/core/domain/mentor_recruitmentdm/budget.go]: ありがとうございます。
なんとなくビジネスルールがあるのであればとりあえず作ったほうがいいのではみたいな考えが自分の中にあったのですが、
大してルールがなかったり使い回すとかがないのであればプリミティブで作るみたいな基準で良いのですね。
ロジックが少なそうなものは値オブジェクトを作らない感じでリファクタリングしてみます！

File: src/core/domain/mentor_recruitmentdm/consultation_method.go (hunk 3 above)
Comment (TakayukiHirano117) [reply src/core/domain/mentor_recruitmentdm/consultation_method.go]: ありがとうございます！
簡単に調べたところpublicの方が良さげな気がするので一旦publicにしてみようかと思います！
publicにしたことで外部から参照されて色々問題でそうであればそれをREADMEに記載するとかで周知できる様にしたいと思います！

File: src/core/domain/mentor_recruitmentdm/budget_test.go (hunk 2 above)
Comment (TakayukiHirano117) [reply src/core/domain/mentor_recruitmentdm/budget_test.go]: @revenue-hack さん
こちらで修正しました！
エラーチェック：[ab2003f](https://github.com/TakayukiHirano117/architecture-study/pull/20/commits/ab2003f56bf5cc79f5000d5a27cde034d7cbfd47)
mentor_recruitmentのテストでプリミティブな値のチェック：[bf89709](https://github.com/TakayukiHirano117/architecture-study/pull/20/commits/bf897096238e869fd9d2d394428a7eb60c3534c7)
